source/grid/cell_id.cc

```diff
@@ -21,13 +21,110 @@
 
 DEAL_II_NAMESPACE_OPEN
 
+
+CellId::CellId(const unsigned int dim,
+               const std::vector<char> &binary_representation)
+{
+  coarse_cell_id = 0;
+  unsigned int i = 0;
+  for (; i<sizeof(unsigned int); ++i)
+    {
+      const unsigned int temp = static_cast<unsigned int> (binary_representation[i]);
+      coarse_cell_id = coarse_cell_id | (temp<<(i*8));
```

Comment:
The C standard does not actually say. There used to be machines with 9 bits per byte but since everyone uses 8 bytes these days, I think it is safe to say that nobody in their right mind would propose a processor these days that does not follow this precedent. I think this is safe.